mdal/frmts/mdal_3di.cpp

```diff
@@ -259,3 +336,135 @@ std::vector<std::pair<double, double>> MDAL::Driver3Di::parseClassification( int
+
+  // Y coordinate
+  int ncidY = mNcFile->getVarId( "Mesh1DNode_ycc" );
+  double fillY = mNcFile->getFillValue( ncidY );
+  std::vector<double> verticesY( vertexCount );
+  if ( nc_get_var_double( mNcFile->handle(), ncidY, verticesY.data() ) ) throw MDAL::Error( MDAL_Status::Err_UnknownFormat, "Unknown format" );
+
+  // Z coordinate
+  int ncidZ = mNcFile->getVarId( "Mesh1DNode_zcc" );
+  double fillZ = mNcFile->getFillValue( ncidZ );
+  std::vector<double> verticesZ( vertexCount );
+  if ( nc_get_var_double( mNcFile->handle(), ncidZ, verticesZ.data() ) ) throw MDAL::Error( MDAL_Status::Err_UnknownFormat, "Unknown format" );
+
+  // Node Id
+  int ncidNodeId = mNcFile->getVarId( "Mesh1DNode_id" );
+  std::vector<int> verticesId( vertexCount );
+  if ( nc_get_var_int( mNcFile->handle(), ncidNodeId, verticesId.data() ) ) throw MDAL::Error( MDAL_Status::Err_UnknownFormat, "Unknown format" );
+
+  // Edge Id
+  int ncidEgeId = mNcFile->getVarId( "Mesh1DLine_id" );
+  std::vector<int> edgesId( edgesCount );
+  if ( nc_get_var_int( mNcFile->handle(), ncidEgeId, edgesId.data() ) ) throw MDAL::Error( MDAL_Status::Err_UnknownFormat, "Unknown format" );
+
+  for ( size_t i = 0; i < vertexCount; ++i )
+  {
+    Vertex vertex;
+    vertex.x = verticesX[i];
+    if ( vertex.x == fillX )
+      vertex.x = std::numeric_limits<double>::quiet_NaN();
+    vertex.y = verticesY[i];
+    if ( vertex.y == fillY )
+      vertex.y = std::numeric_limits<double>::quiet_NaN();
+    vertex.z = verticesZ[i];
+    if ( vertex.z == fillZ )
+      vertex.z = std::numeric_limits<double>::quiet_NaN();
+    vertices[i] = vertex;
+  }
+
+  parse1DConnection( verticesId, edgesId, edges );
+
+  // here we check if there are some edges without
+}
+
+
+bool MDAL::Driver3Di::check1DConnection( std::string fileName )
+{
+  std::string sqliteFile = dirName( fileName ) + "/gridadmin.sqlite";
+
+  if ( !fileExists( sqliteFile ) )
+    return false;
+
+  Sqlite3Db sqliteDb;
+  return sqliteDb.open( sqliteFile );
+
+}
+
+void MDAL::Driver3Di::parse1DConnection( const std::vector<int> &nodesId,
+    const std::vector<int> &edgesId,
+    Edges &edges )
+{
+  std::string sqliteFileName = dirName( mNcFile->getFileName() ) + "/gridadmin.sqlite";
+
+  //construct id map
+  std::map<int, size_t> edgeMap;
+  std::map<int, size_t> nodeMap;
+  for ( size_t edgeIndex = 0; edgeIndex < edges.size(); ++edgeIndex )
+    edgeMap[edgesId.at( edgeIndex )] = edgeIndex;
+
+  for ( size_t nodeIndex = 0; nodeIndex < nodesId.size(); ++nodeIndex )
+    nodeMap[nodesId.at( nodeIndex )] = nodeIndex;
+
+
+  Sqlite3Db db;
+  if ( !db.open( sqliteFileName ) || !( db.get() ) )
+    throw MDAL::Error( MDAL_Status::Err_UnknownFormat, "Unknown format" );
+
+  sqlite3_stmt *stmt;
+  int rc = sqlite3_prepare_v2( db.get(), "SELECT id, start_node_idx, end_node_idx FROM flowlines", -1, &stmt, nullptr );
```

Comment:
encapsulate in mdal_sqlite3.hpp with something like https://github.com/lutraconsulting/geodiff/blob/aa9ffaad0948cdfbf3b07ff992b95b601ff7fe07/geodiff/src/geodiffutils.hpp#L59

for example here I think you have leak here, missing sqlite3_close on stmt?